ocs-api/main.go

```diff
@@ -398,49 +397,71 @@ func postImportKeysHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// Verify content type
-	if httpErr := outils.IsValidPostBinary(r); httpErr != nil {
-		http.Error(w, httpErr.Error(), httpErr.Code)
+	if httpErr := outils.IsValidPostBinary(r); httpErr == nil {
+		http.Error(w, "Error: Passing a key pair tar file into this API is no longer supported.", http.StatusBadRequest)
 		return
 	}
 
-	// Create directory we are putting the file in
-	tarFileDir := "tmp" // in the current dir (home dir)
-	if err := os.MkdirAll(tarFileDir, 0750); err != nil {
-		http.Error(w, "could not create directory "+tarFileDir+": "+err.Error(), http.StatusInternalServerError)
+	if httpErr := outils.IsValidPostJson(r); httpErr != nil {
+		http.Error(w, "Im hoping anything works", httpErr.Code)
 		return
 	}
 
-	// Create a writer that will write the request body to the tar file
-	tarFilePath := tarFileDir + "/owner-keys.tar.gz"
-	f, err := os.Create(tarFilePath)
-	if err != nil {
-		http.Error(w, "could not create file "+tarFilePath+": "+err.Error(), http.StatusInternalServerError)
-		return
+	// Struct for json form containing key pair information
+	type Information struct {
+		Key_name     string `json:"key_name"`
+		Common_name  string `json:"common_name"`
+		Email_name   string `json:"email_name"`
+		Company_name string `json:"company_name"`
+		Country_name string `json:"country_name"`
+		State_name   string `json:"state_name"`
+		Locale_name  string `json:"locale_name"`
 	}
-	fileWriter := bufio.NewWriter(f)
-	_, err = fileWriter.ReadFrom(r.Body)
-	if err != nil {
-		http.Error(w, "could not read request body: "+err.Error(), http.StatusInternalServerError)
-		f.Close()
+
+	info := Information{}
+	// Parse the request body
+	if httpErr := outils.ReadJsonBody(r, &info); httpErr != nil {
+		http.Error(w, httpErr.Error(), httpErr.Code)
 		return
 	}
-	fileWriter.Flush()
-	f.Close()
 
-	// Run the script that will import the tar file keys
-	outils.Verbose("Running command: ./import-owner-private-keys.sh %s %s", tarFilePath, deviceOrgId)
-	stdOut, stdErr, err := outils.RunCmd("./import-owner-private-keys.sh", tarFilePath, deviceOrgId)
+	// Run the script that will create and import the key pairs
+	outils.Verbose("Running command: ./import-owner-private-keys2.sh %s %s %s %s %s %s %s %s", deviceOrgId, info.Key_name, info.Common_name, info.Email_name, info.Company_name, info.Country_name, info.State_name, info.Locale_name) // "%s %s", pemFilePath, deviceOrgId)
+	stdOut, stdErr, err := outils.RunCmd("./import-owner-private-keys2.sh", deviceOrgId, info.Key_name, info.Common_name, info.Email_name, info.Company_name, info.Country_name, info.State_name, info.Locale_name)                    // , pemFilePath, deviceOrgId)
 	if err != nil {
-		http.Error(w, "error running import-owner-private-keys.sh: "+err.Error(), http.StatusBadRequest) // this includes stdErr
+		http.Error(w, "error running import-owner-private-keys2.sh: "+err.Error(), http.StatusBadRequest) // this includes stdErr
```

Comment:
As mentioned in my comments in the script, you should:
- have different exit codes for the different categories of errors the script can hit
- provide a short preface error here, based on the exit code
- then add the stderr from the script to the error msg